### ISSUE TITLE
False positive on undefined-variable in for: else: block

### ISSUE THREAD

**Issue by kbrose** [microsoft/python-language-server#1450]
## Environment data

- VS Code version: 1.37.0
- Extension version (available under the Extensions sidebar): 2019.8.30787
- OS and version: Darwin x64 18.7.0
- Python version (& distribution if applicable, e.g. Anaconda): 3.7.3
- Type of virtual environment used (N/A | venv | virtualenv | conda | ...): pyenv
- Relevant/affected Python packages and their versions: n/a
- Jedi or Language Server? (i.e. what is `"python.jediEnabled"` set to; more info microsoft/vscode-python#3977): `"python.jediEnabled": false`

## Expected behaviour

This code:

```python
for i in range(5):
    pass
else:
    x = 5
    print('padding out', x)
```

should show no errors. Saving this to a file `test.py` and running it raises no errors, and prints the expected string `padding out 5`.

## Actual behaviour

It shows the error

> Undefined variable: 'x' (Python(undefined-variable) [5-26])

despite `x` being a defined variable.

<img width="761" alt="image" src="https://user-images.githubusercontent.com/4053170/63132778-ed0c3080-bf87-11e9-857f-4c379c3ceac8.png">

## Steps to reproduce:
1. Create a file `test.py`
1. Insert code from the "Expected behavior" section of this bug report.
1. Observe that an error is displayed, despite the code being correct.

**Comment by CTrando** [comment]
Fixed in #1423. Can you download the latest version of the language server?

See https://github.com/microsoft/vscode-python/issues/3977

**Comment by MikhailArkhipov** [comment]
Duplicate of #1420. Currently in beta channel, will appear in stable early next week.
